src/rai_extensions/rai_open_set_vision/rai_open_set_vision/vision_markup/segmenter.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (C) 2024 Robotec.AI
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from os import PathLike
+from typing import List, NamedTuple
+
+import hydra
+import numpy as np
+import torch
+from cv_bridge import CvBridge
+from sam2.build_sam import build_sam2
+from sam2.sam2_image_predictor import SAM2ImagePredictor
+from sensor_msgs.msg import Image
+from vision_msgs.msg import BoundingBox2D
+
+from rai.tools.ros.utils import convert_ros_img_to_ndarray
+
+
+class SegmentationResult(NamedTuple):
```

Comment:
Out of curiosity, is there a reason why you used NamedTuple instead of dataclass? (I am not saying it's wrong)

Reply:
Dataclass is an external dependancy, whereas NamedTuple is supported by python standard. Why use a dataclass when a NamedTuple suffices? Also NamedTuple supports decomposition, whereas dataclasses don't.

Reply:
Sure, thank you